packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -908,6 +930,160 @@ class _LargeTitleNavigationBarSliverDelegate
   }
 }
 
+/// The large title of the navigation bar.
+///
+/// Magnifies on over-scroll when [strech] parameter
+/// of the [CupertinoSliverNavigationBar] is true.
+class _LargeTitle extends SingleChildRenderObjectWidget {
+  const _LargeTitle({
+    required this.constraints,
+    required this.maxExtent,
+    required this.textDirection,
+    super.child,
+  });
+
+  final BoxConstraints constraints;
+  final double maxExtent;
+  final TextDirection textDirection;
+
+  @override
+  _RenderLargeTitle createRenderObject(BuildContext context) {
+    return _RenderLargeTitle(
+      navBarConstraints: constraints,
+      maxExtent: maxExtent,
+      textDirection: textDirection,
+    );
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, _RenderLargeTitle renderObject) {
+    renderObject
+      ..navBarConstraints = constraints
+      ..maxExtent = maxExtent
+      ..textDirection = textDirection;
+  }
+}
+
+class _RenderLargeTitle extends RenderProxyBox {
+  _RenderLargeTitle({
+    required BoxConstraints navBarConstraints,
+    required double maxExtent,
+    required TextDirection textDirection,
+  }) : _navBarConstraints = navBarConstraints,
+      _maxExtent = maxExtent,
+      _textDirection = textDirection;
+
+  static const AlignmentDirectional alignment = AlignmentDirectional.bottomStart;
+
+  BoxConstraints get navBarConstraints => _navBarConstraints;
+  BoxConstraints _navBarConstraints;
+  set navBarConstraints(BoxConstraints value) {
+    if (_navBarConstraints == value) {
+      return;
+    }
+    _navBarConstraints = value;
+
+    markNeedsLayout();
+    markNeedsPaint();
```

Comment:
you don't need to mark as needs paint if you mark as needs layout, layout dirtiness always implies paint dirtiness.

Reply:
(3x)